langchain4j-azure-ai-search/src/main/java/dev/langchain4j/store/embedding/azure/search/AzureAiSearchEmbeddingStore.java

```diff
@@ -115,21 +117,32 @@ public Builder index(SearchIndex index) {
             return this;
         }
 
+        /**
+         * If no index is provided, set the name of the default index to be used.
+         *
+         * @param indexName The name of the index to be used.
+         * @return builder
+         */
+        public Builder indexName(String indexName) {
+            this.indexName = indexName;
+            return this;
+        }
+
         public AzureAiSearchEmbeddingStore build() {
             ensureNotNull(endpoint, "endpoint");
```

Comment:
Please have a look at my latest commit at https://github.com/langchain4j/langchain4j/pull/1084/commits/ec7fd326b14f2d6feee66d829cad2beea7f4d0ec where I add a test for this

Reply:
looks good!

Since another comment is resolved now, I'll just my thoughts, however, you decide :)

I think it would be great to think about the following: What if someone else asks you to add one more option, e.g. 'eTag' or anything else?

* how much documentation will you update regarding what erases what and so on?
* how many NULLs / IFs / logical operators will appear in your code?
* will the complexity of the API increase or not?

If you think the builder API will be still OK, then this PR should be merged.

I got another idea of how it could look like from the client perspective. 

```
AzureAiSearchEmbeddingStore.builder()
                .apiKey(properties.apiKey())
                .endpoint(properties.endpoint())
                .dimensions(1000)
//defaultSearchIndex() is a static method provided by AbstractAzureAiSearchEmbeddingStore
                .index(defaultSearchIndex().setName("myName"))
                .build();
```
However, the code above will probably struggle supporting dimensions.

One more approach:

````
AzureAiSearchEmbeddingStore.builder()
                .apiKey(properties.apiKey())
                .endpoint(properties.endpoint())
                .dimensions(1000)
//your implementation provides a template with all fields set, so developers will be able to modify it in the way they want. 
//As a result, the implementation won't suffer from additional NULLs/IF-checks/exceptions thrown.
                .index(indexTemplate -> indexTemplate .setName("myName"))
                .build();
```